Makefile

```diff
@@ -97,12 +97,22 @@ vet:
 			echo "and fix them if necessary before submitting the code for reviewal."; \
 		fi
 
+ifeq ($(OS),Windows_NT)
+INSTALL_GIT_HOOKS := cp .hooks/* .git/hooks/
+else
+INSTALL_GIT_HOOKS := for H in .hooks/*; do ln -sf ../../$$H .git/hooks/; done
+endif
+
+githooks:
+	@if [ -d .git/hooks ]; then $(INSTALL_GIT_HOOKS); fi
```

Comment:
do we definitely want to validate this dir exists already? - You might want to to make one, so windows folks can do the equivalent of `rm -rf .git/hooks && make githooks` if they wanna ensure they have clean state.

Reply:
I don't know why this check exists... It was added explicitly by ed1cbb0a789a2de3f87775444f5ee472278fcad6 so I just wanted to preserve existing behaviour. Yes I agree, I would prefer the Makefile take total control of the .git/hooks dir, so it can be its own make target and cleaned/recreated as one, but not sure if there are some who put their own additional hooks in here for some reason maybe?

Reply:
It's for the reason specified in that commit. If you download the source tarball instead of the git repo and use the Makefile, the .git directory doesn't exist, so the cp would fail and thus the command would fail. If we're not in a git repo we shouldn't add the hooks.

Reply:
I was on my phone when I read this. Yes I eventually did grok the reason for this, which seems totally reasonable, and I am not advocating changing anything in this regard, this PR preserves the existing check.